services/brig/src/Brig/App.hs

```diff
@@ -170,7 +172,9 @@ data Env = Env
     _zauthEnv :: ZAuth.Env,
     _digestSHA256 :: Digest,
     _digestMD5 :: Digest,
-    _indexEnv :: IndexEnv
+    _indexEnv :: IndexEnv,
+    _randomPrekeys :: Bool,
+    _randomPrekeyLocalLock :: MVar ()
```

Comment:
There shouldn't be a local lock for random prekeys if `_randomPrekeys` is `False`.  You could do something like this:

```
    _randomPrekeyLocalLock :: Maybe (MVar ())
```

and then:

```
randomPrekeys :: Getter' Env Bool
randomPrekeys = to $ isJust <$> view randomPrekeyLocalLock
```

In `Brig.Data.Client`, you can then case-switch over the maybe-lock instead of the `ifM`, and you'll have fewer representable bad states.

I think this would be easy, but I'm not sure it's important enough to do it?

Reply:
It's cleaner I agree. On it.